---
  - name: Create instances
    hosts: localhost
    gather_facts: no
    connection: local

    vars:
      machine_type: custom-1-1024  # 1 core, 1.70Gi RAM - got from template created
      image: centos-6
      service_account_email: LONGID-compute@developer.gserviceaccount.com
      # pem_file looks broken on Ansible 2
      # credentials_file was introduced in ansible 2.1.0
      pem_file: /etc/ansible/keys/pkey.pem
      project_id: gce-edge-test
      zone: europe-west1-c

    tasks:
      - name: Launch instances
        gce:
            instance_names: edgetest-1,edgetest-2,edgetest-3,edgetest-4,edgetest-5,edgetest-6,edgetest-7,edgetest-8
            machine_type: "{{ machine_type }}"
            image: "{{ image }}"
            service_account_email: "{{ service_account_email }}"
            pem_file: "{{ pem_file }}"
            project_id: "{{ project_id }}"
            zone: "{{ zone }}"
            tags: gce-test-edges
        register: gce

      - name: Wait for SSH to come up
        wait_for: host={{ item.public_ip }} port=22 delay=5 timeout=100
        with_items: gce.instance_data

      - name: Add host to groupname
        add_host: hostname={{ item.public_ip }} groupname=launched
        with_items: gce.instance_data

  - name: Configure instance(s)
    hosts: launched
    become: yes
    pre_tasks:
      - include_vars: "gce_vars/{{ env }}/vars_provision.yml"
      - name: install selinux bindings 
        yum: name=libselinux-python state=present 
    roles:
      # - geerlingguy.ntp
      # - gce_epel_repo
      - gce_provision_base
      - gce_provision
      - company_cdn_edge
      - gce_end_conf



